nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/oauth2/NuxeoOAuth2Filter.java

```diff
@@ -108,45 +74,16 @@ protected boolean isValid(ServletRequest request) {
         }
 
         HttpServletRequest httpRequest = (HttpServletRequest) request;
-        return isAuthorizedRequest(httpRequest) || httpRequest.getRequestURI().contains(OAUTH2_SEGMENT);
-    }
-
-    protected void process(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException,
-            ServletException {
-        HttpServletRequest httpRequest = (HttpServletRequest) request;
-        HttpServletResponse httpResponse = (HttpServletResponse) response;
-
-        String uri = httpRequest.getRequestURI();
-        if (uri.contains(OAUTH2_SEGMENT)) {
-            String endpoint = uri.split(OAUTH2_SEGMENT)[1];
-
-            switch (endpoint) {
-            case ENDPOINT_AUTH:
-                processAuthorization(httpRequest, httpResponse, chain);
-                break;
-            case ENDPOINT_TOKEN:
-                processToken(httpRequest, httpResponse, chain);
-                break;
-            }
-        } else if (isAuthorizedRequest(httpRequest)) {
-            processAuthentication(httpRequest, httpResponse, chain);
-        }
-
-        if (!response.isCommitted()) {
-            chain.doFilter(request, response);
-        }
+        return isAuthorizedRequest(httpRequest);
     }
 
     protected void processAuthentication(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
             throws IOException, ServletException {
         String key = URLDecoder.decode(request.getHeader("Authorization").substring(7), "UTF-8").trim();
-        NuxeoOAuth2Token token = getTokenStore().getToken(key);
-
-        if (token == null) {
-            return;
-        }
+        NuxeoOAuth2Token token = TransactionHelper.runInNewTransaction(() -> getTokenStore().getToken(key));
```

Comment:
Simply `runInTransaction`. We don't need to start a separate one if one already exists.